cpp/include/cudf/binning/bin.hpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/detail/bin.hpp>
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+
+namespace cudf {
+
+/**
+ * @addtogroup binning
+ * @{
+ * @file
+ * @brief APIs for binning values.
+ */
+
+// TODO: Figure out the best way to document this enum. It needs to be in the
+// detail header to support that API, but that won't be publicly documented and
+// a using declaration doesn't show the values of the enum.
```

Comment:
You could define it here and include the non-detail header in the detail header, right?

Reply:
Yes, I definitely could. At the time it seemed strange to be including a non-detail header in a file whose contents are exclusively in the detail namespace, but I suppose we do that all the time.